arch/x86_64/kernel/memory-manager/primary/primary_phys.cpp

```diff
@@ -0,0 +1,195 @@
+#include "x86_64/memory-manager/primary/primary_phys.hpp"
+
+#include <stl/cstdlib/stdio.h>
+
+#include "x86_64/libk++/align.h"
+#include "x86_64/trace/strace.hpp"
+
+namespace firefly::kernel::mm::primary {
+static libkern::Bitmap bitmap;
+static uint32_t *arena;
+static int64_t linked_list_allocation_base;  //Base address for the linked list structure (Should never be freed, may be reused)
```

Comment:
This should just be a list tho, not a linked list there's no point. but yea

Reply:
like you have 4096 bytes, why not just say "every 8 bytes is an address" and then return it by casting it to a void*[] and a size_t